tools/toolchains/arm.py

```diff
@@ -70,8 +76,12 @@ def __init__(self, target, notify=None, macros=None,
         if target.core not in self.SUPPORTED_CORES:
             raise NotSupportedException(
                 "this compiler does not support the core %s" % target.core)
+            
+        if  getattr(target, "default_lib", "std") == "small" and getattr(target, "microlib_supported", False) == False:
```

Comment:
My understanding is that all targets will always have `default_lib` and `microlib_supported`, if that is true then you can simply write the below instead of always testing with `getattr()`:

```suggestion
        if  target.default_lib == "small" and target.microlib_supported == False:
```

Reply:
this is not true, uARM toolchain supported targets only has default_lib config. Even Microlib_supported config added only to 4 targets,  so i should use getattr

Reply:
Don't all targets in `targets.json` inherits from the `"Target"` attribute?
If that is true then **ALL** targets will have `microlib_supported` and `default_lib` because they have been added to the `"Target"` attribute.

Reply:
I don't think so that's true "Don't all targets in targets.json inherit from the "Target" attribute?" based on existing code `if getattr(target, "default_toolchain", "ARMC6") == "uARM"`

Reply:
Well you can check it right?
The previous code is inconsistent as it does not always check before using attributes of `targets`.

Reply:
It works but in case if default_toolchain is added specifically to target device like K64F in targets.json but for other target devices build tools instead of throwing an exception it uses getattr with default returns(3rd argument) for default_toolchain. I prefer to go with the same approach what build tools follow.

Reply:
`getattr` is to be used if you are unsure that the object has the attribute which is not the case here as you are guaranteed to have the attributes. Because something pointless already exist does not mean you have to follow it.

Reply:
Instead of getting inherit "microlib_supported" and "default_lib" to all target device and I will only add those 2keyword to specific targets and go with getattr

Reply:
Why? That defeats the purpose. All targets must have them and they should have default values `False` and `std` to have a fail safe.

Reply:
This is not for all targets, only supported targets but I can do this for default_lib and microlib_supported and not for uARM get attr.

Reply:
We cannot guarantee that `microlib_supported` is set if it comes from targets.json. What if the parameter is inadvertently removed from targets.json?